File: examples/tcp_server_example.rb
#!/usr/bin/env ruby

require 'hurricane'

gateway = Hurricane::Gateway.new(
  Erlang::SocketWrapper.new('localhost', 3000))
gateway.register_server('time_server')
loop do
  request = gateway.do_recv()

  response = Hurricane::Message.new()
  response.type = 'response'
  response.destination = request.destination
  response.tag = request.tag
  response.data = Time.new.inspect()

  gateway.do_send(response)
end

File: examples/tcp_server_example.py
#!/usr/bin/env python

from datetime import datetime
from hurricane import Gateway, Message
from hurricane.erl_codec import SocketWrapper, Atom

def main():
    gateway = Gateway(SocketWrapper('localhost', 3000))
    gateway.register_server('time_server')
    while True:
        request = gateway.recv()
        response = Message()
        response.type = 'response'
        response.destination = request.destination
        response.tag = request.tag
        response.data = str(datetime.now())
        gateway.send(response)

if __name__ == '__main__':
    main()
